### PR TITLE
feat: find dbtproject subdirectory

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #2204

### Description:
cli should find dbt_project.yml if I'm in a subdirectory
By recursively checking its parent until / 

### Preview:

Screenshot from 2022-05-24 11-05-17

if can't find it

Too many logs ? I could only show it once...
image

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD  
- [x] CLI

### CODE REVIEW

**Review by ZeRego — APPROVED** [review]

File: packages/cli/src/dbt/context.ts [hunk 1]
```diff
@@ -15,16 +16,33 @@ export type DbtContext = {
...
     const projectFilename = path.join(projectDir, 'dbt_project.yml');
     let config;
+
     try {
         config = yaml.load(
             await fs.readFile(projectFilename, { encoding: 'utf-8' }),
         ) as any;
-    } catch (e) {
+    } catch (e: any) {
+        if (projectDir !== '/') {
```
Comment (owlas) [inline packages/cli/src/dbt/context.ts:29]: This isn't platform dependent. I think `os` or `path` will give you an option to check if you're at the root directory

File: packages/cli/src/dbt/context.ts [hunk 2]
```diff
@@ -15,16 +16,33 @@ export type DbtContext = {
...
+            const parentDir = path.join(projectDir, '..');
+            console.log(
```
Comment (owlas) [inline packages/cli/src/dbt/context.ts:31]: I don't think the user needs to know this. If it's never found you could add to the error message that `dbt_project.yml` was not found in this directory or any of it's parent directories

File: packages/cli/src/dbt/manifest.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const loadManifest = async ({ targetDir }: LoadManifestArgs) => {
             await fs.readFile(filename, { encoding: 'utf-8' }),
         ) as any;
         return manifest;
-    } catch (err) {
+    } catch (err: any) {
```
Comment (owlas) [inline packages/cli/src/dbt/manifest.ts:15]: why was this needed?

**Review by owlas — CHANGES_REQUESTED** [review]
nice recursion, see inline comments

File: packages/cli/src/dbt/manifest.ts (hunk 3 above)
Comment (rephus) [reply packages/cli/src/dbt/manifest.ts:15]: I got these errors when building 

image

Comment (ZeRego) [reply packages/cli/src/dbt/manifest.ts:15]: In one of the latest Typescript releases they changed the error type from `any` to `unknown`.  Do you have a different version installed locally or have we changed the version recently ? 🤔

Comment (ZeRego) [reply packages/cli/src/dbt/manifest.ts:15]: It is part of 4.4.0 [(release notes)](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#using-unknown-in-catch-variables) and we use `"typescript": "^4.2.3"` for the cli package but `"typescript": "^4.5.5",` for the frontend package so your IDE might be using the latest version for all the project ignoring the version at the workspace level

Comment (rephus) [reply packages/cli/src/dbt/manifest.ts:15]: I installed this version manually while I was trying to fix the conflicts on CLI. I resintalled all node_modules and I got  typescript: '^4.2.3' version now, now the `unknown` doesn't happen. 

Still, this is not a breaking change, so I'd suggest to leave it as it is
